### DISCUSSION TITLE
WriteableBitmap issue

### DISCUSSION THREAD

**Discussion by vadimaus** [AvaloniaUI/Avalonia#11814] [Q&A] [answered]
Is WriteableBitmap supposed to work in a Console application?

With the the latest stable Avalonia 0.10.21 I get: System.NullReferenceException: 'Object reference not set to an instance of an object.'
For the latest RC 11.0.0-rc1.1 the error is: System.InvalidOperationException: 'Unable to locate 'Avalonia.Platform.IPlatformRenderInterface'.' 

To reproduce the issue: Create a console app and create an instance of WriteableBitmap
WriteableBitmap wb = new WriteableBitmap(new PixelSize(100, 100), new Vector(96, 96), PixelFormat.Bgra8888, AlphaFormat.Premul);

Your help is much appreciated!

**Comment by kekekeks** [comment, marked as answer]
Avalonia APIs aren't usable until you complete the setup through the AppBuilder. You can use headless platform and `SetupWithoutStarting()` to initialize Avalonia without access to the window system.
